swarm/src/connection/pool.rs

```diff
@@ -692,17 +691,45 @@ where
                     let check_peer_id = || {
                         if let Some(peer) = expected_peer_id {
                             if peer != obtained_peer_id {
-                                return Err(PendingConnectionError::WrongPeerId {
-                                    obtained: obtained_peer_id,
-                                    endpoint: endpoint.clone(),
-                                });
+                                return match &endpoint {
+                                    ConnectedPoint::Dialer { address, .. } => {
+                                        Err(PoolEvent::PendingOutboundConnectionError {
+                                            id,
+                                            error: PendingOutboundConnectionError::WrongPeerId {
+                                                obtained: obtained_peer_id,
+                                                address: address.clone(),
+                                            },
+                                            peer: expected_peer_id.or(Some(obtained_peer_id)),
+                                        })
+                                    }
+                                    ConnectedPoint::Listener {.. } => unreachable!("There shouldn't be an expected PeerId on inbound connections."),
+                                };
                             }
                         }
 
                         if self.local_id == obtained_peer_id {
-                            return Err(PendingConnectionError::LocalPeerId {
-                                endpoint: endpoint.clone(),
-                            });
+                            return match &endpoint {
+                                ConnectedPoint::Dialer { address, .. } => {
+                                    Err(PoolEvent::PendingOutboundConnectionError {
+                                        id,
+                                        error: PendingOutboundConnectionError::LocalPeerId {
+                                            address: address.clone(),
+                                        },
+                                        peer: expected_peer_id.or(Some(obtained_peer_id)),
```

Comment:
```suggestion
                                        peer: Some(obtained_peer_id),
```

We already checked that `expected_peer_id` is either `None` or  same as `obtained_peer_id`.

Reply:
Nvm just saw your above comment.